webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/services/DisplayDirectiveTree.scala

```diff
@@ -134,6 +136,7 @@ object DisplayDirectiveTree extends Loggable {
     , keepCategory    : FullActiveTechniqueCategory => Boolean = _ => true
     , keepTechnique   : FullActiveTechnique => Boolean = _ => true
     , keepDirective   : Directive => Boolean = _ => true
+    , userService     : UserService
```

Comment:
in the web part, we don't pass services like that, they are private fields if the snippet, like `linkUtil` above (line 114)